### ISSUE TITLE
spurios syntax help on onehost delete 

### ISSUE THREAD

**Issue by FlorianHeigl** [OpenNebula/one#2254]
# Bug Report
## Version of OpenNebula

- [x] 5.2.2
- [ ] 5.4.0
- [ ] 5.4.1
- [ ] 5.4.2
- [ ] 5.4.3
- [ ] 5.4.4
- [ ] 5.4.5
- [ ] 5.4.6
- [ ] 5.4.7
- [ ] 5.4.8
- [ ] 5.4.9
- [ ] 5.4.10
- [ ] 5.4.11
- [ ] 5.4.12
- [ ] 5.4.13
- [ ] 5.4.14
- [ ] 5.4.15
- [ ] Development build

## Component

- [ ] Authorization (LDAP, x509 certs...)
- [ ] Command Line Interface (CLI)
- [ ] Contextualization
- [ ] Documentation
- [ ] Federation and HA
- [ ] Host, Clusters and Monitorization
- [ ] KVM
- [ ] Networking
- [ ] Orchestration (OpenNebula Flow)
- [ ] Packages
- [ ] Scheduler
- [ ] Storage & Images
- [ ] Sunstone
- [ ] Upgrades
- [ ] User, Groups, VDCs and ACL
- [ ] vCenter

## Description

So, with a badly performing oned we first get an unhandled ReadTimeout, and then comes the more interesting part: turns out the first command succeeded in the end.
So the host ID is not wrong - but the command line was successfully parsed.
Due to a bug in the parser and function call it then proceeds to try deleting after it stripped the missing host from the command line. Meaning it calls the delete with 0 hosts to delete!!!

```
flo.ifz-nw11(M)nodeae6b:~$ onehost delete node85fd

Net::ReadTimeout
flo.ifz-nw11(M)nodeae6b:~$ 
flo.ifz-nw11(M)nodeae6b:~$ onehost delete node85fd
HOST named node85fd not found.
command delete: argument 0 must be one of range, hostid_list

```

**Comment by rsmontero** [comment]
It is not actually making the call to oned. Note that it refers to argument 0 not to host id 0. 

BTW, you may benefit to increase the timeout for client API calls and/or CLI with environment `ONE_XMLRPC_TIMEOUT`

So this looks as a works for me

**Comment by FlorianHeigl** [comment]
@rsmontero
- it refers to argument 0 because that's where it passes the ID?
- it calls the function that would delete, not calls to oned
- It should not give the syntax help after stripping the single argument (missing node)

sorry but please don't try to convince me this is good enough :)

**Comment by rsmontero** [comment]
Let me rephrase this. The expected behavior would be:

```
flo.ifz-nw11(M)nodeae6b:~$ onehost delete node85fd
HOST named node85fd not found.
```

**without** the `command delete: ...` message?

**Comment by FlorianHeigl** [comment]
Yes :)
